Enable clang-tidy on master

1ntEgr8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61674
* #61711
* __->__ #61689

Differential Revision: [D29767984](https://our.internmc.facebook.com/intern/diff/D29767984)

1ntEgr8 committed:
Enable clang-tidy on master

[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

driazati commented on .github/workflows/lint.yml:
```diff
@@ -345,6 +345,21 @@ jobs:
+      # Run clang-tidy on a smaller subset of the codebase on master until we
+      # make the repository clang-tidy clean
+      - name: Run clang-tidy
+        if: github.event_name == 'push'
```
nit: limit to `master` since people push to other pytorch branches all the time but we don't care about those

```suggestion
        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

driazati commented on .github/workflows/lint.yml:
```diff
@@ -345,6 +345,21 @@ jobs:
+            --paths torch/csrc/fx \
+            --clang-tidy-exe "$(which clang-tidy)" \
+            --diff-file pr.diff \
+            --verbose \
```
nit: match previous invocation

```suggestion
```

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 committed:
Update on "Enable clang-tidy on master"




[ghstack-poisoned]

1ntEgr8 commented:
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29767984).

1ntEgr8 committed:
Update on "Enable clang-tidy on master"


Differential Revision: [D29767984](https://our.internmc.facebook.com/intern/diff/D29767984)

[ghstack-poisoned]

1ntEgr8 commented:
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29767984).

1ntEgr8 committed:
Update on "Enable clang-tidy on master"


Differential Revision: [D29767984](https://our.internmc.facebook.com/intern/diff/D29767984)

[ghstack-poisoned]

1ntEgr8 commented:
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29767984).
